### ISSUE TITLE
Enhance `unused_param` with strict enforcement for unexported functions

### ISSUE THREAD

**Issue by denisvmedia** [mgechev/revive#1657]
**Is your feature request related to a problem? Please describe.**

Consider:

```go
func boo(a int, b string) {
	println(a)
}
```

`b` is unused.

In unexported functions this is very often just leftover from a refactor. The code compiles, existing `unused_param` logic may allow renaming to `_`, and the dead parameter remains in place.

This request is **not** about detecting unused parameters in general. That already exists.

The problem is that in some cases we don’t want to allow renaming to `_` at all — we want to forbid unused parameters entirely for certain kinds of functions (e.g. unexported ones), because they are almost always accidental and represent signature drift.

At the same time, we must not break legitimate callback use cases where the full signature is required.

**Describe the solution you'd like**

Extend `unused_param` with a strict mode (e.g. a flag) that:

* Ignores exported functions and methods.
* For unexported functions:
  * Reports unused parameters as errors.
  * Does not allow `_` or `_name` as a workaround.
  * Skips reporting if the function is used as a value (e.g. passed as a callback).
* Optionally applies to unexported methods (behind a flag), due to potential interface usage.

In strict mode, the expectation is: if the parameter is unused and the function is internal and not part of a functional contract, it must be removed.

**Describe alternatives you've considered**

* Relying on `_` renaming — hides the issue instead of removing dead parameters.
* Creating a separate linter — unnecessary duplication of `unused_param` logic.
* Globally forbidding unused parameters — too strict for real-world callback patterns.
* Attempting full interface implementation detection — too complex and unreliable for local analysis.

**Additional context**

The goal is to prevent silent accumulation of dead parameters in internal code, especially after refactors or AI-assisted edits.

This is intended as a stricter mode of `unused_param`, not a replacement for it.

**Comment by denisvmedia** [comment]
@chavacava @alexandear wdyt about this rule extension?

**Comment by alexandear** [comment]
Thanks for the detailed description. This is a good idea and can be implemented.

**Comment by chavacava** [comment]
I agree with not proposing/allowing `_` as parameter name in non-exported functions.

I'm not convinced with adding a flag. I do not see why it's necessary. If the rule evolves to:
1. do not allow `_`  as name for not-exported functions, and 
2. stops proposing `_` as name for unused parameters of non-exported functions

then it copes with the problem you describe.
